### PR TITLE
Separate WaitForWorkItemCompletion and FailOnWorkItem/TestFailure

### PR DESCRIPTION
**Pull request by alexperovich**

### CODE REVIEW

File: src/Microsoft.DotNet.Helix/Sdk/Helpers.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+using System.Reflection;
+
+namespace Microsoft.DotNet.Helix.Sdk
+{
+    public static class Helpers
+    {
+        public static ProductInfoHeaderValue UserAgentHeaderValue =>
+            new ProductInfoHeaderValue(new ProductHeaderValue("HelixSdk", ProductVersion));
+
+        public static string ProductVersion =>
```
Comment (ChadNedzlek) [inline src/Microsoft.DotNet.Helix/Sdk/Helpers.cs]: This is a relatively expensive call. Do you want to just cache it with ` { get; } ` instead of ` => `?

Comment (alexperovich) [reply src/Microsoft.DotNet.Helix/Sdk/Helpers.cs]: yep, fixing

File: src/Microsoft.DotNet.Helix/Sdk/HelixWait.cs [hunk 2]
```diff
@@ -27,26 +27,27 @@ public class HelixWait : HelixTask
         [Required]
         public string Build { get; set; }
 
-        public bool IsExternal { get; set; } = false;
```
Comment (jonfortescue) [inline src/Microsoft.DotNet.Helix/Sdk/HelixWait.cs]: Removing IsExternal bc it's no longer necessary in the new way of doing things, right? Can you make sure you update the documentation [here](https://github.com/dotnet/arcade/blob/master/Documentation/AzureDevOps/SendingJobsToHelix.md) accordingly and mark the parameter as deprecated in the template [here](https://github.com/dotnet/arcade/blob/master/eng/common/templates/steps/send-to-helix.yml)?

File: src/Microsoft.DotNet.Helix/Sdk/Readme.md [hunk 3]
```diff
@@ -88,6 +88,23 @@ Given a local folder `$(TestFolder)` containing `runtests.cmd`, this will run `r
     <HelixTargetQueues>Windows.10.Amd64.Open;Ubuntu.1604.Open</HelixTargetQueues>
 
 
+    <!--
```
Comment (jonfortescue) [inline src/Microsoft.DotNet.Helix/Sdk/Readme.md]: THANK I forgot to update this

File: src/Microsoft.DotNet.Helix/Sdk/HelixWait.cs (hunk 2 above)
Comment (alexperovich) [reply src/Microsoft.DotNet.Helix/Sdk/HelixWait.cs]: Yep. Ill update those.

File: src/Microsoft.DotNet.Helix/Sdk/Readme.md (hunk 3 above)
Comment (alexperovich) [reply src/Microsoft.DotNet.Helix/Sdk/Readme.md]: Yea, I have been lazy in updating this. I just added everything we've added.

Comment (jonfortescue) [reply src/Microsoft.DotNet.Helix/Sdk/Readme.md]: I mean don't blame yourself it's my job to update it to lol 💁‍♂️ oops

**Pull request merged**